### PR TITLE
feat: #1001 add NoAuth_NavMenu_Close and NoAuth_NavMenu_Home_Language tests

### PR DESCRIPTION
**Pull request by EkaterinaSokolova-V**
Tests added:
NoAuth_NavMenu_Close
NoAuth_NavMenu_Home_Language

Comment:
Not sure about 177 line in mw-webapp/src/component/select/Select.tsx

### CODE REVIEW

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts [hunk 1]
```diff
@@ -53,5 +54,30 @@ describe('Navigation menu scope tests', () => {
         cy.url().should('include', aboutProjectPageData.endpoint);
         aboutProjectSelectors.welcomeBlock.getTitle().should('contain', aboutProjectPageContent.mainTitle.en);
     });
+
+    it('NoAuth_NavMenu_Close', () => {
+        cy.get('body').click(345, 10, {force: true});
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/navigationMenu.cy.ts:59]: Why 345, 10? What will be if design will bee changed?

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts [hunk 2]
```diff
@@ -53,5 +54,30 @@ describe('Navigation menu scope tests', () => {
         cy.url().should('include', aboutProjectPageData.endpoint);
         aboutProjectSelectors.welcomeBlock.getTitle().should('contain', aboutProjectPageContent.mainTitle.en);
     });
+
+    it('NoAuth_NavMenu_Close', () => {
+        cy.get('body').click(345, 10, {force: true}); 
+
+        cy.url().should('include', '/');
+        navigationMenuSelectors.menuItemLinks.getAboutProjectItemLink().should('not.exist');
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/navigationMenu.cy.ts:62]: This test check is navMenu close if click on button? If yes then why on 62 line we check only existing  AboutProjectItem link?

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts [hunk 3]
```diff
@@ -53,5 +54,30 @@ describe('Navigation menu scope tests', () => {
...
+    });
+
+    it('NoAuth_NavMenu_Home_Language', () => {
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/navigationMenu.cy.ts:65]: NoAuth_NavMenu_Home_Language => NoAuth_NavMenu_Language

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts [hunk 4]
```diff
@@ -53,5 +54,30 @@ describe('Navigation menu scope tests', () => {
...
+        languageCode.forEach((code, index) => {
+            navigationMenuSelectors.language.languageMenuItems[index]().should('have.text', code);
+            navigationMenuSelectors.language.languageMenuItems[index]().click();
+    
+            navigationMenuSelectors.language.getText().should('have.text', languageText[index]);
+
+            if (index < languageText.length - 1) {
+                navigationMenuSelectors.language.getSelect().click();
+              }
+        });
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/navigationMenu.cy.ts:80]: why need to click if this test check languages? Maybe I don't understand something, then ok(

File: mw-webapp/src/component/header/Header.tsx [hunk 5]
```diff
@@ -31,10 +31,10 @@ import styles from "src/component/header/Header.module.scss";
...
 export const languageOptions: SelectItemType<Language>[] = [
-  {id: "1", value: Language.ENGLISH, text: "EN"},
-  {id: "2", value: Language.RUSSIAN, text: "RU"},
-  {id: "3", value: Language.UKRAINIAN, text: "UA"},
-  {id: "4", value: Language.GEORGIAN, text: "KA"},
+  {id: "1", value: Language.ENGLISH, text: "EN", dataCy: navigationMenuIds.language.enItem},
+  {id: "2", value: Language.RUSSIAN, text: "RU", dataCy: navigationMenuIds.language.ruItem},
+  {id: "3", value: Language.UKRAINIAN, text: "UA", dataCy: navigationMenuIds.language.uaItem},
+  {id: "4", value: Language.GEORGIAN, text: "KA", dataCy: navigationMenuIds.language.kaItem},
 ];
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.tsx:38]: We have enum Language. Is it possibe to use it here for consistency?

File: mw-webapp/src/component/select/Select.tsx [hunk 6]
```diff
@@ -24,6 +24,11 @@ export interface SelectItemType<T> {
    * Option`s visible text
    */
   text: string;
+
+    /**
+     * Data attributes for cypress testing
+     */
+    dataCy?: string;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:31]: Please check spaces

File: mw-webapp/src/component/select/Select.tsx [hunk 7]
```diff
@@ -142,6 +147,11 @@ export interface SelectProps<T> {
    * Custom class name
    */
   className?: string;
+
+    /**
+     * Data attributes for cypress testing
+     */
+  dataCy?: string;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:154]: 1. PLease check spaces
2/ We have already cy prop for SelectProps interface. Is it really needed to use addinition dataCy prop?

File: mw-webapp/src/component/select/Select.tsx [hunk 8]
```diff
@@ -163,7 +174,7 @@ export const Select = <T extends string>(props: SelectProps<T>) => {
   return (
     <div
       className={clsx(styles.select, props.className)}
-      data-cy={props.cy?.dataCyOverlay}
+      data-cy={`${props.cy?.dataCyOverlay || ""} ${props.dataCy || ""}`.trim()}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:177]: Looks a bit strange. Please check  is it right?

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 1 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts:59]: I can use any element of the page to click on it. May be it's better in this case?

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 2 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts:62]: This test checks whether the menu closes after clicking on any part of the page outside the menu. 
in this case AboutProjectItem link is checked if there is no link from the menu - no menu.

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 4 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts:80]: I switch languages ​​sequentially and check that the language changes in the selector and menu.

File: mw-webapp/src/component/select/Select.tsx (hunk 7 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/select/Select.tsx:154]: If I delete it I get an error, I don’t know how to use the existing one.

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 3 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 8 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/select/Select.tsx]: fixed

File: mw-webapp/src/component/select/Select.tsx (hunk 6 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/select/Select.tsx]: dataCy removed

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts]: Let's find header by data-cy attribute, to avoid mistakes in the future )

Comment (Ekaterina1994) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts]: Также закрывать меню можно кликая вне сайдбара, вроде бы была раньше такая опция

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts:62]: Мне кажется, что правильнее проверять, именно закрывается ли сайдбар, а не отсутствие ссылок? Что если в дальнейшем, например, ссылки будут пропадать, но меню не закрываться? )

File: mw-webapp/src/component/select/Select.tsx [hunk 9]
```diff
@@ -24,6 +24,11 @@ export interface SelectItemType<T> {
    * Option`s visible text
    */
   text: string;
+
+  /**
+   * Data attributes for cypress testing
+   */
+    dataCy?: string;
 }
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:32]:
```suggestion
  /**
   * Data attributes for cypress testing
   */
  dataCy?: string;
}
```

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 2 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts:62]: I completely agree, I don’t understand where in the code to add the data cy attribute for the navigation menu.

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 1 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts]: Changed for header.

File: mw-webapp/cypress/e2e/navigationMenu.cy.ts (hunk 2 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/navigationMenu.cy.ts]: @Ekaterina1994 , done

**Review by Ekaterina1994 — APPROVED** [review]

**Pull request merged**